modules/@apostrophecms/modal/ui/apos/components/AposModalShareDraft.vue

```diff
@@ -0,0 +1,211 @@
+<template>
+  <AposModal
+    :modal="modal"
+    class="apos-share-draft"
+    v-on="{ esc: close }"
+    @no-modal="$emit('safe-close')"
+    @inactive="modal.active = false"
+    @show-modal="modal.showModal = true"
+  >
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <div class="apos-share-draft__header">
+            <h2 class="apos-share-draft__heading">
+              {{ $t('apostrophe:shareDraftHeader') }}
+            </h2>
+            <Close
+              class="apos-share-draft__close"
+              :title="$t('apostrophe:close')"
+              :size="18"
+              @click.prevent="close"
+            />
+          </div>
+          <div class="apos-share-draft__content">
+            <div class="apos-share-draft__toggle-wrapper">
+              <AposToggle
+                v-model="disabled"
+                class="apos-share-draft__toggle"
+                @toggle="toggle"
+              />
+              <p class="apos-share-draft__toggle-label">
+                {{ $t('apostrophe:shareDraftEnable') }}
+              </p>
+            </div>
+            <p class="apos-share-draft__description">
+              {{ $t('apostrophe:shareDraftDescription') }}
+            </p>
+            <input
+              type="text"
+              value="https://dummy-url.com/wefe3456wfewuidfki9wefe3456wf554ewfki9"
+              disabled
+              class="apos-share-draft__url"
+            >
+            <a
+              href=""
+              class="apos-share-draft__link-copy"
+              @click.prevent="copy"
+            >
+              <LinkVariant
+                class="apos-share-draft__link-icon"
+                :title="$t('apostrophe:shareDraftCopyLink')"
+                :size="16"
+              />
+              &nbsp;{{ $t('apostrophe:shareDraftCopyLink') }}
+            </a>
+          </div>
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+import Close from 'vue-material-design-icons/Close.vue';
+import LinkVariant from 'vue-material-design-icons/LinkVariant.vue';
+
+export default {
+  components: {
+    Close,
+    LinkVariant
+  },
+  emits: [ 'safe-close' ],
+  data() {
+    return {
+      modal: {
+        active: false,
+        type: 'overlay',
+        showModal: false,
+        disableHeader: true,
+        trapFocus: true
+      },
+      disabled: false
+    };
+  },
+  async mounted() {
+    this.modal.active = true;
+  },
+  methods: {
+    copy() {
+      console.log('copy');
+    },
+    toggle() {
+      console.log('toggle');
+      this.disabled = !this.disabled;
+    },
```

Comment:
we can keep these logs until the actual implementation is done in https://linear.app/apostrophecms/issue/PRO-2815/as-a-contributor-editor-admin-i-can-share-and-un-share-documents-via